catalog/core/catalog-core-definitionparser/src/main/java/ddf/catalog/definition/impl/DefinitionParser.java

```diff
@@ -366,6 +384,16 @@ private List<Callable<Boolean>> parseMetacardTypes(
               changeset.metacardValidatorServices.add(registration);
               return registration != null;
             });
+
+        staged.add(
+            () -> {
+              requiredAttributesRegistry.addRequiredAttributes(
```

Comment:
So, I think this is fine, but it does mean that we will only find required attributes if they were defined in the definition json files.  If any other bundle registers their own required attributes validator directly, we wouldn't know about them. 
 I'm wondering if we should instead make the `RequiredAttributesMetacardValidator` extend a new `RequiredAttributes` interface and then you could listen for the registering and unregistering of those instances to build up the set of required attributes.  Or, we could create a RequiredAttributes interface that anyone could register and then we could change the implementation of the RequiredAttributesMetacardValidator to be a "singleton" rather than having several of them registered.  This code here would then just create those RequiredAttributesImpl objects instead of an actual RequiredAttributesMetacardValidator.  The registry that you created would then just listen for RequiredAttributes instances being registered and unregistered to update its internal map.